cmd/rofl/build/validate.go

```diff
@@ -132,23 +160,40 @@ func validateComposeFile(composeFile string, manifest *buildRofl.Manifest) error
 				}
 			}
 
-			return fmt.Errorf("volume '%s:%s' of service '%s' has an invalid external source (should be '/run/rofl-appd.sock', '/run/podman.sock' or reside inside '/storage/')", vol.Source, vol.Target, serviceName)
+			return nil, fmt.Errorf("volume '%s:%s' of service '%s' has an invalid external source (should be '/run/rofl-appd.sock', '/run/podman.sock' or reside inside '/storage/')", vol.Source, vol.Target, serviceName)
 		}
 
 		// Validate ports.
 		publishedPorts := make(map[uint64]struct{})
 		for _, port := range service.Ports {
 			if port.Target == 0 {
-				return fmt.Errorf("service '%s' has an invalid zero port defined", serviceName)
+				return nil, fmt.Errorf("service '%s' has an invalid zero port defined", serviceName)
 			}
 			if port.Published == "" || port.Published == "0" {
-				return fmt.Errorf("port '%d' of service '%s' does not have an explicit published port defined", port.Target, serviceName)
+				return nil, fmt.Errorf("port '%d' of service '%s' does not have an explicit published port defined", port.Target, serviceName)
 			}
 			publishedPort, err := strconv.ParseUint(port.Published, 10, 16)
 			if err != nil {
-				return fmt.Errorf("published port '%s' of service '%s' is invalid: %w", port.Published, serviceName, err)
+				return nil, fmt.Errorf("published port '%s' of service '%s' is invalid: %w", port.Published, serviceName, err)
 			}
 			publishedPorts[publishedPort] = struct{}{}
+
+			if port.Protocol != "tcp" {
+				continue
+			}
+			proxyMode := service.Annotations[fmt.Sprintf("net.oasis.proxy.ports.%s.mode", port.Published)]
+			if proxyMode == "ignore" {
+				continue
+			}
+			genericDomain := fmt.Sprintf("p%s", port.Published)
+			customDomain := service.Annotations[fmt.Sprintf("net.oasis.proxy.ports.%s.custom_domain", port.Published)]
```

Comment:
is it wise to put the domain name inside `compose.yaml` instead of `rofl.yaml`? Suppose you have a testing and a production version of your app. Then you'd need two separate compose files in this case.

Reply:
Hm, yeah I wanted to simplify the amount of configuration for the user.

It definitely needs to be set in `compose.yaml` because you can have a domain for each port and it makes sense to have that configuration in one place.

But what you said also makes sense. Maybe we should just have per-deployment artifacts?

Reply:
No that would currently not work as the parser is not doing environment variable substitution when parsing the YAML compose file.